client-extensions/liferay-object-collection-filter-extension/src/common/services/liferay/api.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Liferay} from './liferay.js';
+
+const {REACT_APP_LIFERAY_HOST = window.location.origin} = process.env;
+
+const baseFetch = async (url, options = {}) => {
+	return fetch(REACT_APP_LIFERAY_HOST + '/' + url, {
+		headers: {
+			'Content-Type': 'application/json',
+			'Authorization': 'Basic ' + btoa('test@liferay.com:1')
```

Comment:
I don't think using user test here is a good way, my recommendation is 
`import axios, { AxiosRequestConfig, AxiosResponse } from 'axios'
import { Liferay } from './liferay'

const REACT_APP_LIFERAY_HOST: string = window.location.origin

const baseFetch = async (
  method: string,
  url: string,
  body?: any,
  options: AxiosRequestConfig = {}
): Promise<AxiosResponse> => {
  try {
    const response = await axios({
      method,
      url: `${REACT_APP_LIFERAY_HOST}/${url}`,
      headers: {
        'Content-Type': 'application/json',
        'x-csrf-token': Liferay.authToken,
        ...options.headers,
      },
      data: body,
      ...options,
    })

    return response
  } catch (error) {
    throw error
  }
}

export default baseFetch
`

Reply:
OK